src/olympia/users/models.py

```diff
@@ -95,7 +101,150 @@ def build_widget_attrs(self, *args, **kwargs):
         return attrs
 
 
+class UserQuerySet(BaseQuerySet):
+    def ban_and_disable_related_content(self):
+        """Admin method to ban multiple users and disable the content they
+        produced.
+
+        Similar to deletion, except that the content produced by the user is
+        forcibly soft-disabled instead of being deleted where possible, and the
+        user is not anonymized: we keep their data until hard-deletion kicks in
+        (see clear_old_user_data), including fxa_id and email so that they are
```

Comment:
Not a change in this pr... but we need `clear_old_user_data` to be able to hard delete instances but the queryset delete methods have been changed to soft-delete instead. (Not applicable to UserProfile or Addon either because we ban and disable those respectively rather than delete)

Reply:
Yeah, I'm surprised no tests broke because of that. I've filed https://github.com/mozilla/addons-server/issues/21482 for that.